src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php

```diff
@@ -54,13 +54,48 @@ public function testVote(array $attributes, $expectedVote, $object, $message)
 
         $this->assertEquals($expectedVote, $voter->vote($this->token, $object, $attributes), $message);
     }
+
+    /**
+     * @dataProvider getTests
+     * @group legacy
+     */
+    public function testVoteLegacy(array $attributes, $expectedVote, $object, $message)
+    {
+        $voter = new AbstractVoterTest_LegacyVoter();
+
+        $this->assertEquals($expectedVote, $voter->vote($this->token, $object, $attributes), $message);
+    }
+
+    /**
+     * @group legacy
+     * @expectedException \BadMethodCallException
+     */
+    public function testNoOverriddenMethodsThrowsException()
+    {
+        $voter = new AbstractVoterTest_NothingImplementedVoter();
+        $voter->vote($this->token, new \stdClass(), array('EDIT'));
+    }
 }
 
 class AbstractVoterTest_Voter extends AbstractVoter
 {
+    protected function voteOnAttribute($attribute, $object, TokenInterface $token)
+    {
+        return 'EDIT' === $attribute;
+    }
+
+    protected function supports($attribute, $class)
+    {
+        return $this->isClassInstanceOf($class, 'stdClass')
+            && in_array($attribute, array('EDIT', 'CREATE'));
+    }
+}
+
+class AbstractVoterTest_LegacyVoter extends AbstractVoter
+{
     protected function getSupportedClasses()
     {
-        return array('stdClass');
+        return array('AbstractVoterTest_Object');
```

Comment:
Afaics, this should fix the tests (but I'm on a mobile now, sp can't be 100% sure).